documentation/extension.tex

```diff
@@ -1 +1,98 @@
 \chapter{VS Code extension}
+
+The frontend of the project is implemented as an extension to a modern IDE instead of creating a completely new GUI. This approach has the advantage of providing a familiar environment and workflow that the developers are used to.
+
+There are several IDEs that currently (natively) support LSP, such as \emph{Eclipse Che} and \emph{Eclipse IDE}, \emph{vim8}, \emph{Visual Studio} and \emph{Visual Studio Code} and many more. Others, for example \emph{IntelliJ}, have plugins which add the support for LSP.
+
+Our IDE choice is \emph{Visual Studio Code}, due to its popularity and lightweight design. Conveniently, \emph{Theia}, a web-based IDE, supports VSCode extensions, therefore our plugin works with \emph{Theia} as well.
+
+\section{Standard LSP Extension}
+
+The core of the extension is an activation event which starts the plugin for VSCode.
+
+Upon activation, \emph{Language Client} and \emph{Language server}~\ref{chap:lang_server} are started as child processes of VSCode and a pipe is open for their communication. The LSP communication and its features are handled by the \emph{vscodelc} package. 
+
+To be independent of pipes, we have added an option to use TCP, which assigns a random free port for TCP communication.
+
+\section{DAP Extension}
+
+\emph{Macro Tracer}~\ref{chap:macro_tracer} is implemented using DAP, which is also supported out-of-the-box by VSCode. We dynamically assign a random free port for DAP communication during the activation as well.
```

Comment:
We dynamically assign a random free port for DAP communication during the activation to support more language servers running at he same time?

To as well mi tam nesedi, tvari sa to ako keby tam boli 2 veci co supportujeme.

Reply:
to as well je odkaz na paragraph vyssie. nechcel som pouzit rovnaku vetu totiz. preformulujem